src/cargo/core/resolver/encode.rs

```diff
@@ -30,6 +32,42 @@ struct Patch {
 pub type Metadata = BTreeMap<String, String>;
 
 impl EncodableResolve {
+    /// Try to preserve equivalence over version changes in Cargo.
+    ///
+    /// For example, even with [root] being present in one and not in the other,
+    /// the semantic meaning is still the same, so return true.
+    pub fn semantically_eq(self, other: Self) -> bool {
```

Comment:
I am somewhat afraid that these needs to be updated every time we add new field to `EncodableResolve`.

Perhaps we should have `fn normalize(&mut self)`, and then use natural equality?

Reply:
I agree with @matklad in that we may want to pursue alternative means if we can, for example can `into_resolve` be used below and then we would equate two instances of `Resolve`?